services/management/dbaas/xtradb_cluster_service.go

```diff
@@ -88,13 +89,24 @@ func (s XtraDBClusterService) ListXtraDBClusters(ctx context.Context, req *dbaas
 			}
 		}
 
-		if c.Params.Proxysql.ComputeResources != nil {
-			cluster.Params.Proxysql = &dbaasv1beta1.XtraDBClusterParams_ProxySQL{
-				DiskSize: c.Params.Proxysql.DiskSize,
+		if c.Params.Proxysql != nil {
+			if c.Params.Proxysql.ComputeResources != nil {
+				cluster.Params.Proxysql = &dbaasv1beta1.XtraDBClusterParams_ProxySQL{
+					DiskSize: c.Params.Proxysql.DiskSize,
+					ComputeResources: &dbaasv1beta1.ComputeResources{
+						CpuM:        c.Params.Proxysql.ComputeResources.CpuM,
+						MemoryBytes: c.Params.Proxysql.ComputeResources.MemoryBytes,
+					},
+				}
 			}
-			cluster.Params.Proxysql.ComputeResources = &dbaasv1beta1.ComputeResources{
-				CpuM:        c.Params.Proxysql.ComputeResources.CpuM,
-				MemoryBytes: c.Params.Proxysql.ComputeResources.MemoryBytes,
+		} else {
```

Comment:
we can do away with the else here too, the same way as this https://github.com/percona-platform/dbaas-controller/pull/171#discussion_r609676070

Reply:
I added the check for `haproxy != nil` just to be sure. Again I use continue to skip the other proxy when first one is set.